airbyte-integrations/connectors/source-shopify/source_shopify/streams/streams.py

```diff
@@ -93,6 +95,10 @@ def request_params(
             params["status"] = "any"
         return params
 
+    def get_error_handler(self) -> Optional[ErrorHandler]:
+        default_handler = super().get_error_handler()
+        return LimitReducingErrorHandler(stream=self, default_handler=default_handler)
```

Comment:
As mentioned below, I think it makes more sense to subclass `LimitReducingErrorHandler` instead of instantiating an inner error handler like we do here. Since the error handler we create in the base `ShopifyStream` just uses default arguments or other fields we already have access to:
```
def get_error_handler(self) -> Optional[ErrorHandler]:
        known_errors = ShopifyNonRetryableErrors(self.name)
        error_mapping = DEFAULT_ERROR_MAPPING | known_errors
        return LimitReducingErrorHandler(self.logger, max_retries=5, error_mapping=error_mapping, stream=self)
```

Something like that. And then the `LimitReducingErrorHandler.__init__` can either invoke the parent `super()` constructor or set the fields accordingly